docs.md

```diff
@@ -1,23 +1,396 @@
-`allow {
-   path = /collections/...
-}`
-`deny {
-   path = /collections/lakes
-}`
--  Handling equality check between a property on the request object and a parameter
\ No newline at end of file
+View the Architecture in the browser [Here](https://www.figma.com/file/684S7kO4dPQbbZFZr6xZOn/Rego-builder?node-id=0%3A1) 
+
+
+
+   Installation
+   ============
+- Manual installation - Available [Here](https://github.com/r-scheele/rego_builder#readme)
+
+- Build your own image and start the container -
+   ```bash
+   $ Configure environment variables - the .env file
+   $ docker-compose -f docker-compose.dev.yml up -d
+
+   ```
+- Pull the official image from Docker Hub - and start the container 
+
+   ```bash
+   $ docker pull rscheele3214/rego-builder:latest
+   $ docker run rscheele3214/rego-builder:latest
+   ```
+
+
+How the translation from JSON to REGO works?
+============
+The REGO policies are created by a set of functions that serves as the translation logic, They're called commands. Each of these commands is responsible for writing specific REGO rule to a `.rego` file. However, the JSON to REGO conversion must follow certain conditions as defined by a pydantic model for effectiveness. The following is an example of how a rule is defined:
+
+   ```json     
+   {
+      "command": "input_prop_equals",
+      "properties": {
+         "input_property": "request_method",
+         "value": "GET"
+      }
+   }
+   ```
+In this case, The command key represents the operation to be performed and the properties key represents the properties that are being used in the operation.
+`input_prop_equals` is the command in example, which initiates the appropriate operation, on the properties object.
+The above rule translates to an equality check between the input_property and the value. <br />
+The REGO equivalent of the above rule object is: <br />
+   `input.request_method == "GET"`
+
+### As of now, the API supports the following commands:
+1. input_prop_equals
+2. input_prop_in
+3. input_prop_in_as
+4. allow_full_access
+
+
+Usage
+============
+### Detailed explanation of commands with examples: 
+
+
+## 1. Input_props_equals
+   This command has different logic to handle series of equality checks.
+  ### - Handling '*' as the wildcard flag: <br /> 
+
+   This logic handles all the paths after a particular section. if `/collections/` is supplied as the option, all the routes after it will be allowed e.g allow `/collections/obs/`, allow `/collections/test-data/obs/`, allow `/collections/obs/` etc. <br /> 
+
+   Rule is simply written as: <br />
+   `allow { 
+      input.request_path = /collections/...
+   }` <br />  and the rule object is:
+   ```json     
+   {
+      "command": "input_prop_equals",
+      "properties": {
+         "input_property": "request_path",
+         "value": ["v1", "collections", "*"]
+      }
+   }
+   ```
```

Comment:
Rephrase.

Example:

This logic handles all the paths after a particular section. Let's look at an example rule:

---json file---

In the JSON above, the `value` key contains an asterik in the values section. This indicates that the endpoint x is allowed to do y.

---rego equivalent---

Same applies for every other command in this section.